lib/slim-rails/register_engine.rb

```diff
@@ -0,0 +1,30 @@
+module Slim
+  module Rails
+    module RegisterEngine
+      def self.register_engine(app, config)
+        if ::Rails::VERSION::MAJOR == 3
+          register_engine3(app, config)
+        else
+          register_engine(app, config)
+        end
+      end
+
+      private
+
+      def self.register_engine3(app, _config)
+        return unless app.assets
+        return unless app.assets.respond_to?(:register_engine)
+
+        app.assets.register_engine('.slim', Slim::Template)
+      end
+
+      def self.register_engine(_app, config)
```

Comment:
There are same methods name already, i'll rename it `register_engine4_or_later`.